### PR TITLE
feature/jdyer_HiggsPortalKDIFee

### PR DESCRIPTION
**Pull request by jamiecd**
This feature branch contains the generation-level fcl files needed for my November production requests.

### CODE REVIEW

File: fcl/gen/text/textfilegen_HiggsScalar_M100_th2e-4_KDIF.fcl [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+# ------------------------------------------------------------------------------
+physics: {
+
+  producers: {
+    
+    # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+    
+    generator: {
+      module_type:    TextFileGen
+      InputFileName: "/icarus/app/users/jdyer/scalargen/my_dev_area_placeTextFiles/srcs/icarus_data/icarus_data/HiggsPortal/100MeV_mySim_unweighted_master.txt"
```
Comment (jzennamo) [inline fcl/gen/text/textfilegen_HiggsScalar_M100_th2e-4_KDIF.fcl:96]: This won't be accessible on the grid, could you have this moved to `icarus_data` (by contacting @SFBayLaser, Tracy Usher) and verify that you can find the TextFile with this module

File: fcl/gen/text/textfilegen_HiggsScalar_M150_th2e-4_KDIF.fcl [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+# ------------------------------------------------------------------------------
+physics: {
+
+  producers: {
+    
+    # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+    
+    generator: {
+      module_type:    TextFileGen
+      InputFileName: "/icarus/app/users/jdyer/scalargen/my_dev_area_placeTextFiles/srcs/icarus_data/icarus_data/HiggsPortal/150MeV_mySim_unweighted_master.txt"
```
Comment (jzennamo) [inline fcl/gen/text/textfilegen_HiggsScalar_M150_th2e-4_KDIF.fcl:96]: This won't be accessible on the grid, could you have this moved to `icarus_data` (by contacting @SFBayLaser, Tracy Usher) and verify that you can find the TextFile with this module

File: fcl/gen/text/textfilegen_HiggsScalar_M200_th2e-4_KDIF.fcl [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+# ------------------------------------------------------------------------------
+physics: {
+
+  producers: {
+    
+    # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+    
+    generator: {
+      module_type:    TextFileGen
+      InputFileName: "/icarus/app/users/jdyer/scalargen/my_dev_area_placeTextFiles/srcs/icarus_data/icarus_data/HiggsPortal/200MeV_mySim_unweighted_master.txt"
```
Comment (jzennamo) [inline fcl/gen/text/textfilegen_HiggsScalar_M200_th2e-4_KDIF.fcl:96]: This won't be accessible on the grid, could you have this moved to `icarus_data` (by contacting @SFBayLaser, Tracy Usher) and verify that you can find the TextFile with this module

File: fcl/gen/text/textfilegen_HiggsScalar_M50_th2e-4_KDIF.fcl [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+# ------------------------------------------------------------------------------
+physics: {
+
+  producers: {
+    
+    # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+    
+    generator: {
+      module_type:    TextFileGen
+      InputFileName: "/icarus/app/users/jdyer/scalargen/my_dev_area_placeTextFiles/srcs/icarus_data/icarus_data/HiggsPortal/50MeV_mySim_unweighted_master.txt"
```
Comment (jzennamo) [inline fcl/gen/text/textfilegen_HiggsScalar_M50_th2e-4_KDIF.fcl:96]: This won't be accessible on the grid, could you have this moved to `icarus_data` (by contacting @SFBayLaser, Tracy Usher) and verify that you can find the TextFile with this module

**Review by jzennamo — CHANGES_REQUESTED** [review]
Just to verify, have you checked that when you run many jobs you get different entries in your text file? I thought that in the setup you provided you'll always get the first N entries in your text file, where N is the number of events you generate per job.

**Comment by jamiecd** [comment]
> Good catch - you're correct that you'll always get the first N entries in the text file. I don't know of a way to specify which N entries, so that a job does not have to start at the beginning of the text file, but let me do some digging. 

Alternatively, since this phase is very fast, is it possible to run all 12500 events for this phase with one job?

**Comment by jzennamo** [comment]
> Alternatively, since this phase is very fast, is it possible to run all 12500 events for this phase with one job?

I don't think that this will work since when we move to the next production steps (G4, detsim, reco) we can only support 20-50 events per job.

File: fcl/gen/text/textfilegen_HiggsScalar_M100_th2e-4_KDIF.fcl (hunk 1 above)
Comment (jamiecd) [reply fcl/gen/text/textfilegen_HiggsScalar_M100_th2e-4_KDIF.fcl:96]: I will need [this feature branch](https://cdcvs.fnal.gov/redmine/projects/icarus_data/repository/show/icarus_data/HiggsPortal?rev=feature%2Fjdyer_HiggsPortalKDIFee) of icarus_data merged in order to access my Input Files. Tracy says I need to state that request here, in this PR... Can you merge that branch of icarus_data for me?

**Comment by jamiecd** [comment]
> > Alternatively, since this phase is very fast, is it possible to run all 12500 events for this phase with one job?
> 
> I don't think that this will work since when we move to the next production steps (G4, detsim, reco) we can only support 20-50 events per job.

I've now run this first stage myself, and have root files ready to be fed into G4 in my `/pnfs` directory. For each mass point, I have a file with all 12500 events in addition to smaller files containing 10 events each in case that is preferred for splitting across jobs. Hopefully not having to worry about his first stage will simplify things for Production. Please let me know what you'd like me to do from here.

**Comment by SFBayLaser** [comment]
This appears to have an outstanding change and the two reviewers need to sign off. Can this be done today?

**Comment by SFBayLaser** [comment]
Also, am I correct that this also needs to have the feature branch https://cdcvs.fnal.gov/redmine/projects/icarus_data/repository/show/icarus_data/HiggsPortal?rev=feature%2Fjdyer_HiggsPortalKDIFee merged into icarus_data as well?

**Comment by jamiecd** [comment]
Hi Tracy, There's been a lot of offline discussion about my PR since my last comment 11 days ago (eg following the slides I presented at the PWF meeting last Thursday: https://sbn-docdb.fnal.gov/cgi-bin/sso/RetrieveFile?docid=24230&filename=21.12.09_SBN_ProductionWorkflow_Meeting-Dyer.pdf&version=1 ). This pull request/branch was for my fcl files that convert text files into root files that can be processed by G4 and the rest of the production chain. They are no longer needed since I have now run that stage on my own. For the same reason, the icarus_data branch you linked above does not need to be merged anymore either.

**Comment by SFBayLaser** [comment]
Closing this pull request as no longer needed

**Pull request closed without merging**